### ISSUE TITLE
LSP does not support 'C'

### ISSUE THREAD

**Issue by DhruvaKashyap** [hlissner/doom-emacs#5364]
### What did you expect to happen?

When I add 
```
(cc +lsp)
```
I expect autocomplete using ccls/clangd via LSP for C,C++,Obj-C

### What actually happened?

LSP does not support a backend for autocomplete for 'C'. 

### Steps to reproduce

1. Open Emacs
2. Add ```(cc +lsp)``` to the ~/.doom.d/init.el
3. Install ccls/clangd
4. Restart Emacs after doom sync
5. Open a C file
6. Wait for autocomplete after typing ```prin```

### System Information

https://pastebin.com/5bNGrLeJ

**Comment by DhruvaKashyap** [comment]
I would like to add that, since there's only a global cc mode, it would not be possible to configure irony-server/rtags for C and keep LSP for C++.

**Comment by hlissner** [comment]
Can you be more specific about how "LSP does not support a backend"? How did you determine this? Any errors/messages in the \*Messages\* or \*lsp-log\* buffers?

It appears to work fine for me, out of the box, in my SDL projects.

I'd wager that Emacs can't find your ccls/clangd executables. If this is the case, run `$ doom env` on the command line and restart Emacs.

**Comment by DhruvaKashyap** [comment]
I forgot to add this link, here https://emacs-lsp.github.io/lsp-mode/page/languages/, where they do not show any support for C.
Which is what prompted me to believe that LSP does not support a backend for C. More specifically, they do not state that ccls/clangd works even with C.

I don't have large projects so I usually don't use a .ccls file.

LSP works fine with C++, it was able to find the clangd executable.

I just ran ```$doom env``` , and it has started to perform autocomplete for C, I have been scratching my head about this for quite a while.  I don't understand why it would work on C++ files but not C files.
The issue has been resolved for me, but I am unsure as it how it arose in the first place.
Thank you for Doom!
